### PR TITLE
Add handling when span id and trace id are undefined

### PR DESCRIPTION
**Pull request by ida613**
### What does this PR do?
When we only propagate [baggage](https://github.com/DataDog/dd-trace-js/pull/4563) and nothing else, trace id and span id could be undefined. This PR adds handling for this case. According to otel specification, baggages are attached to contexts rather than spans, but contexts are tied to spans in dd-trace-js. We are essentially using spans to propagate baggages, so trace id and span id could be missing.

### Plugin Checklist

- [ ] Unit tests.
- [ ] TypeScript [definitions][1].
- [ ] TypeScript [tests][2].
- [ ] API [documentation][3].
- [ ] CircleCI [jobs/workflows][4].
- [ ] Plugin is [exported][5].

[1]: https://github.com/DataDog/dd-trace-js/blob/master/index.d.ts
[2]: https://github.com/DataDog/dd-trace-js/blob/master/docs/test.ts
[3]: https://github.com/DataDog/documentation/blob/master/content/en/tracing/trace_collection/library_config/nodejs.md
[4]: https://github.com/DataDog/dd-trace-js/blob/master/.circleci/config.yml
[5]: https://github.com/DataDog/dd-trace-js/blob/master/packages/dd-trace/src/plugins/index.js

### CODE REVIEW

File: packages/dd-trace/src/opentracing/span_context.js [hunk 1]
```diff
@@ -36,13 +36,15 @@ class DatadogSpanContext {
         ? this._trace.tags[TRACE_ID_128] + this._traceId.toString(16).padStart(16, '0')
         : this._traceId.toString(16).padStart(32, '0')
     }
+    if (this._traceId === undefined) return 'unavailable'
```
Comment (rochdev) [inline packages/dd-trace/src/opentracing/span_context.js:39]: Why not just handle this in the caller? (or even better not rely on the span context for this)

Also, this is not tested.

**Review by rochdev — CHANGES_REQUESTED** [review]

File: packages/dd-trace/src/opentracing/span_context.js (hunk 1 above)
Comment (ida613) [reply packages/dd-trace/src/opentracing/span_context.js:39]: can we rely on the user to do exception handling? They shouldn't call these two functions if they are only propagating baggage, but I'm not sure if I can trust them

Comment (ida613) [reply packages/dd-trace/src/opentracing/span_context.js:39]: on a second thought you are probably right, I'll close this PR

Comment (rochdev) [reply packages/dd-trace/src/opentracing/span_context.js:39]: If they are using OTel they would never be using our internal `SpanContext`.

**Pull request closed without merging**